### PR TITLE
move rendering marker hover to hover widget

### LINKED ISSUE #62370: Error hover semantic information should be rendered separate of error message
From @octref in #62159

> This is a regression on the diagnostics display.
> 
> 1.28:
> 
> image
> 
> Insiders:
> 
> image
> 
> In both case, the diagnostics returned are:
> 
> ```js
> const diagnostics = {
>   code: "unknownProperties",
>   source: "css.lint.unknownProperties",
>   message: "Unknown property: 'foo1'"
> }
> ```

If we have access to semantic information we should render it much better. The message should come first. Then the source on a separate line, with a `Source:` label preceding it and in non-monospace font (since source should be a user-readable string like `CSS Lint`). Then, the code with the same style, but with monospace, since code is often a library/OS error code. Something like:

> `Unknown property: 'foo1'`
>
> **Source:** CSS Lint
> **Code:** `unknownProperties`

cc @sandy081  @ramya-rao-a @misolori

### PR DESCRIPTION
**Pull request by sandy081**
Fix #62370 

Moved rendering marker hover into hover widget which can display marker message and details nicely. And also applies right editor font to the message.

### CODE REVIEW

File: src/vs/base/common/htmlContent.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ export class MarkdownString implements IMarkdownString {
 
 	value: string;
 	isTrusted?: boolean;
-	sanitize: boolean = true;
```
Comment (alexandrudima) [inline src/vs/base/common/htmlContent.ts]: Same, I don't understand this change.

File: src/vs/editor/common/services/modelServiceImpl.ts [hunk 2]
```diff
@@ -283,7 +119,7 @@ export class ModelServiceImpl extends Disposable implements IModelService {
 	constructor(
 		@IMarkerService markerService: IMarkerService | null,
```
Comment (alexandrudima) [inline src/vs/editor/common/services/modelServiceImpl.ts]: I think `IMarkerService` is no longer needed in this class, I would remove it.

File: src/vs/base/browser/htmlContentRenderer.ts [hunk 3]
```diff
@@ -211,7 +211,7 @@ export function renderMarkdown(markdown: IMarkdownString, options: RenderOptions
 	}
 
 	const markedOptions: marked.MarkedOptions = {
-		sanitize: markdown instanceof MarkdownString ? markdown.sanitize : true,
```
Comment (alexandrudima) [inline src/vs/base/browser/htmlContentRenderer.ts]: I don't understand this part...

**Review by alexdima — APPROVED** [review]
:+1: Looks very good to me.

I don't understand a couple of changes and I would remove the `IMarkerService` from `ModelServiceImpl` and adopt in tests.

File: src/vs/base/common/htmlContent.ts (hunk 1 above)
Comment (sandy081) [reply src/vs/base/common/htmlContent.ts]: `sanitize` flag was a hack added before to render diagnostic message by generating and passing html through markdown string. Used this flag not to sanitize the html by the markdown renderer. With this change this flag is no longer needed and hence removed it.

File: src/vs/editor/common/services/modelServiceImpl.ts (hunk 2 above)
Comment (sandy081) [reply src/vs/editor/common/services/modelServiceImpl.ts]: `IMarkerService` is still being used by `ModelServiceImpl` to remove the markers of models with internal URI scheme. I have no issues to move that clean up to new service and remove `IMarkerService` in `ModelServiceImpl`.

Comment (sandy081) [reply src/vs/editor/common/services/modelServiceImpl.ts]: Done

**Pull request merged**